### ISSUE TITLE
Rust std faster to decode utf8

### ISSUE THREAD

**Issue by tafia** [hsivonen/encoding_rs#7]
On my system `std::str::from_utf8` and `std::string::String::from_utf8` are both faster than encoding_rs `UTF_8` (on the file I'm testing, not sure this is the general case).
Is there a reason not to defaults to std?

Benches here:
https://github.com/tafia/bench_decode

**Comment by hsivonen** [comment]
Good discovery! Thank you.

I verified that the difference wasn't just difference in memory copy overhead by [adding validation-only tests to your code](https://github.com/hsivonen/bench_decode) and by rerunning [my own benchmark](https://github.com/hsivonen/encoding_bench).

It turns out that a [change](https://github.com/rust-lang/rust/commit/408c2f7827be838aadcd05bd041dab94388af35d#diff-667f8101ac1fe1b2f63b7c2ef0691502) has landed to std after I forked the UTF-8 validation code from there to encoding_rs and made it faster in encoding_rs. I'll need to see if those changes can be integrated into encoding_rs or if it makes sense to just unfork.

The reason not to default to std was that forking the code (at least temporarily) allowed out-of-std experimentation with SIMD.
